pages/product.js

```diff
@@ -120,70 +128,97 @@ class ProductPage extends React.PureComponent{
+            <img
+              className="thumbnail_big"
+              ref="displayImage"
+              src={this.props.Item.imageUrl}
+            />
             <div>
-              <img className="thumbnail_small"
-                src={router.query.image}
-                onMouseOver={this.changeImage} />
-              <img className="thumbnail_small"
+              <img
+                className="thumbnail_small"
+                src={this.props.Item.imageUrl}
+                onMouseOver={this.changeImage}
+              />
+              <img
+                className="thumbnail_small"
                 src="/images/upay.png"
-                onMouseOver={this.changeImage} />
-              <img className="thumbnail_small"
-                src={router.query.image}
-                onMouseOver={this.changeImage} />
-              <img className="thumbnail_small"
-                src={router.query.image}
-                onMouseOver={this.changeImage} />
+                onMouseOver={this.changeImage}
+              />
+              <img
+                className="thumbnail_small"
+                src={this.props.Item.imageUrl}
+                onMouseOver={this.changeImage}
+              />
+              <img
+                className="thumbnail_small"
+                src={this.props.Item.imageUrl}
+                onMouseOver={this.changeImage}
+              />
             </div>
           </div>
 
           <div className="description_container">
             <div className="product_info_container">
-              <h1>{router.query.name}</h1>
-              {router.query.dprice ? (
+              <h1>{this.props.Item.name}</h1>
+              {this.props.Item.originalPrice ? (
                 <s>
                   <Rupee />
-                  {router.query.price}
+                  {this.props.Item.originalPrice}
                 </s>
-              ):(
+              ) : (
                 <span>
                   <Rupee />
-                  {router.query.price}
+                  {this.props.Item.originalPrice}
                 </span>
               )}
-              {router.query.dprice && (
+              {this.props.Item.discountedPrice && (
                 <span>
                   <Rupee />
-                  <strong>{router.query.dprice}</strong>
+                  <strong>{this.props.Item.discountedPrice}</strong>
                 </span>
               )}
               <p>Product description here</p>
             </div>
             <div className="quantity_container">
-              <button className="increment_button" onClick={this.increment}>+</button>
+              <button className="increment_button" onClick={this.increment}>
+                +
+              </button>
               <div className="quantity">{this.state.qty}</div>
-              <button disabled={!this.state.qty} className="decrement_button" onClick={this.decrement}>-</button>
+              <button
```

Comment:
It'll be nice if you separate these small UI components as individual components along with their CSS. It'll tidy things up and separate out the logic.

Reply:
Don't worry about conflicts, they'll happen anyway.